httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyAsyncResponseListener.java

```diff
@@ -103,6 +103,11 @@ public void onContent(Response response, ByteBuffer content, Callback callback)
     }
   }
 
+  @Override
+  public void onFailure(Response response, Throwable failure) {
+    asyncResponse.completeExceptionally(failure);
```

Comment:
I can't remember, but I do recall using the Adapter to avoid implementing all methods.
It's certainly strange that the onFailure wasn't implemented before. However, the HttpClient behavior did change a little after the original implementation (especially the retries part).

Anyway, it does seem OK to have this.

Reply:
It looks like overriding onComplete ensured timely completion - but does not have appropriate exception handling. 

What we are probably looking for here instead is using onComplete Result.getRequestFailure - that maps most closely to these types of exceptions. There is also Result.getResponseFailure which should be the same as what is passed to onFailure.

Reply:
Oh this gets fun 😅 

Updating `onComplete` to respect `Result.isSucceeded()` or `Result.isFailed()` causes [the expectContinue](https://github.com/SamBarker/kubernetes-client/blob/25ecf8460eb132ae2945342b33f64a4240a2b91c/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/http/AbstractHttpPostTest.java#L130) test to fail, because Jetty considers a 404 in response to a request with `Expect=100-continue` to be a [failure](https://github.com/jetty/jetty.project/blob/2881fe9ab569946f36ec9dac6c59205b4b969007/jetty-client/src/main/java/org/eclipse/jetty/client/ContinueProtocolHandler.java#L84). So i've tried adding `server.expect().post().withPath("/post-expect-continue").andReply(100, request -> "").always();` which causes the server to reply appropriately to the `POST` request. However that breaks `expectContinue` for `jetty`, `OkHttp` and `VertX` (there doesn't seem to be a JDK version 👀 ) clients tests 😢 .

In the jetty case the test now fails because the response is considered pending and there is no further `demand` so the request future never completes. I haven't debugged the others to work out whats going wrong but I suspect its something similar. 

Reading [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/100) on `100-continue` suggests the jetty implementation is wrong as it attaches the body to the `POST` request at the same time as the `Expect=100-continue` header. Whereas MDN suggests that it should send `Expect=100-continue` and wait for the response to  that before continuing by sending the body. 

I'm nothing like close enough to the details of the Kube API server to understand if 100-continue is commonly used or if it has a problem with including the `POST` body on the initial request. However this is starting to feel like a rabbit hole and not really in scope for this PR and thus what the implications of this are in practice. 

Thoughts @manusa, @shawkins ?

Reply:
Ahh the JDK client explicitly disables the `expectContinue` test.

Reply:
> Reading [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/100) on 100-continue suggests the jetty implementation is wrong as it attaches the body to the POST request at the same time as the Expect=100-continue header. Whereas MDN suggests that it should send Expect=100-continue and wait for the response to that before continuing by sending the body.

After further debugging the jetty Impl is right, jetty handles delaying the body if the Expect header is set but I'm still not seeing the request complete.

Reply:
Hmm turns out I was very wrong about what was going on after much debugging I have 
https://github.com/fabric8io/kubernetes-client/pull/6197/files 

Which changes the default socketPolicy for the mockWebserver to `EXPECT_CONTINUE` from `KEEP_OPEN`. I haven't found a way to make it conditional on the headers as its too late by the time dispatch is called. I'm not at all sure what implications that change has.

I'm happy to merge that PR into this one if preferred but I think they are really independent. I'm happy to clean up that PR and its description and or raise an issue to cover it if the consensus is to keep it independent.

Reply:
Nice work, I think it's better to keep them separate.